ts/features/bonus/bpd/saga/orchestration/optInPaymentMethods/optInDeletionChoiceHandler.ts

```diff
@@ -0,0 +1,38 @@
+import { Effect, put, take } from "redux-saga/effects";
+import { ActionType, getType, isActionOf } from "typesafe-actions";
+import { deleteAllPaymentMethodsByFunction } from "../../../../../../store/actions/wallet/delete";
+import { EnableableFunctionsEnum } from "../../../../../../../definitions/pagopa/EnableableFunctions";
+import { bpdUpdateOptInStatusMethod } from "../../../store/actions/onboarding";
+import { CitizenOptInStatusEnum } from "../../../../../../../definitions/bpd/citizen_v2/CitizenOptInStatus";
+
+/**
+ * This saga orchestrate the choice of the user to delete the payment methods added during the cashback.
+ * This saga execute 2 actions:
+ * - delete all the payment methods with the BPD capability
+ * - store the user choice
+ */
+
+export function* optInDeletionChoiceHandler(): Generator<Effect, void, any> {
+  // Perform the payment methods deletion
+  yield put(
+    deleteAllPaymentMethodsByFunction.request(EnableableFunctionsEnum.BPD)
+  );
+  const deleteAllPaymentMethodsByFunctionStatus: ActionType<
+    | typeof deleteAllPaymentMethodsByFunction.success
+    | typeof deleteAllPaymentMethodsByFunction.failure
+  > = yield take([
+    getType(deleteAllPaymentMethodsByFunction.success),
+    getType(deleteAllPaymentMethodsByFunction.failure)
+  ]);
+  if (
+    isActionOf(
+      deleteAllPaymentMethodsByFunction.success,
+      deleteAllPaymentMethodsByFunctionStatus
+    )
+  ) {
+    // If the payment methods deletion succeeded, perform the opt-in update
+    yield put(
+      bpdUpdateOptInStatusMethod.request(CitizenOptInStatusEnum.DENIED)
```

Comment:
```suggestion
      bpdUpdateOptInStatusMethod.request(CitizenOptInStatusEnum.ACCEPTED)
```
If I well understood the flow
❓ is this action  also dispatched by `OptInPaymentMethodsThankYouKeepMethodsScreen.tsx`?
Could it happen it will be dispatched twice in this flow?

Reply:
Yes the action is dispatched also in the `OptInPaymentMethodsThankYouKeepMethodsScreen.tsx`.

I'm not sure I've understood well your second question.
If the action is dispatched in the `OptInPaymentMethodsThankYouKeepMethodsScreen` this saga is not triggered.
So I exclude it can be dispatched twice in the flow.